src/Language/Mimsa/Store/Storage.hs

```diff
@@ -88,61 +78,14 @@ saveExpr se = saveExpr' (se $> ())
 saveExpr' :: StoreExpression () -> StoreM ExprHash
 saveExpr' expr = do
   storePath <- liftIO getExpressionFolder
-  let json = JSON.encode expr
-  let exprHash = getHash json
+  let (json, exprHash) = coerce $ contentAndHash expr
   liftIO $ BS.writeFile (filePath storePath exprHash) json
   pure exprHash
 
 findExpr ::
-  [ServerUrl] ->
-  ExprHash ->
-  StoreM (StoreExpression ())
-findExpr urls hash = findExprInLocalStore hash <|> tryServers hash urls
-
-tryServers ::
```

Comment:
This remote server stuff is worth having, but this is a messy proof of concept that is just complicating things for now.